### PR TITLE
Fix deprecation warning about promise.defer()

### PR DESCRIPTION
**Pull request by MTRNord**
webpack3 started to complain about promise.defer() being deprecated

### CODE REVIEW

File: src/languageHandler.js [hunk 1]
```diff
@@ -231,35 +231,30 @@ export function getCurrentLanguage() {
...
-
-    let response_return = {};
-    request(
-        { method: "GET", url: langPath },
-        (err, response, body) => {
-            if (err || response.status < 200 || response.status >= 300) {
-                deferred.reject({err: err, response: response});
-                return;
+    return new Promise((resolve, reject) => {
+        let response_return = {};
```
Comment (lukebarnard1) [inline src/languageHandler.js]: This is unused

**Comment by lukebarnard1** [comment]
Also, this needs sign-off

File: src/languageHandler.js (hunk 1 above)
Comment (MTRNord) [inline src/languageHandler.js]: Hm yeah it was already unused before. I will clean it up

**Comment by MTRNord** [comment]
Signed-Off by Marcel Radzio <mtrnord1@gmail.com>

**Comment by MTRNord** [comment]
Uhm why do my tests fail. Related to old develop base?

**Comment by lukebarnard1** [comment]
>Related to old develop base?

The tests are done against the current state of `develop`, as is the merge itself.

**Pull request merged**